src/mem3/include/mem3.hrl

```diff
@@ -12,12 +12,12 @@
 
 % type specification hacked to suppress dialyzer warning re: match spec
 -record(shard, {
-    name :: binary() | '_',
-    node :: node() | '_',
-    dbname :: binary(),
-    range :: [non_neg_integer() | '$1' | '$2'] | '_',
+    name :: binary() | '_' | 'undefined',
```

Comment:
For `ref` the order is `'undefined' | '_'`, for others is `'_' | 'undefined'`. Other than that looks good to me.